stl/inc/span

```diff
@@ -226,29 +226,38 @@ struct pointer_traits<_Span_iterator<_Ty>> {
 };
 
 // STRUCT TEMPLATE _Span_extent_type
-template <size_t _Extent>
+template <class _Ty, size_t _Extent>
 struct _Span_extent_type {
+    using pointer = _Ty*;
+
     constexpr _Span_extent_type() noexcept = default;
 
-    constexpr explicit _Span_extent_type(size_t) noexcept {}
+    constexpr explicit _Span_extent_type(size_t, const pointer _Data) noexcept : _Mydata{_Data} {}
 
     _NODISCARD constexpr size_t size() const noexcept {
         return _Extent;
     }
+
+protected:
+    pointer _Mydata{nullptr};
 };
 
-template <>
-struct _Span_extent_type<dynamic_extent> {
+template <class _Ty>
+struct _Span_extent_type<_Ty, dynamic_extent> {
+    using pointer = _Ty*;
+
     constexpr _Span_extent_type() noexcept = default;
 
-    constexpr explicit _Span_extent_type(const size_t _Size) noexcept : _Mysize(_Size) {}
+    constexpr explicit _Span_extent_type(const size_t _Size, const pointer _Data) noexcept
+        : _Mysize(_Size), _Mydata{_Data} {}
 
     _NODISCARD constexpr size_t size() const noexcept {
         return _Mysize;
     }
 
-private:
+protected:
     size_t _Mysize{0};
+    pointer _Mydata{nullptr};

```

Comment:
Does the order matter for performance? If we access `_Mydata` frequently, is it important for it to be at offset 0?

Note: I don't like being constrained by anything when it comes to choosing our internal representations (especially "why can't you use the same internal representation as another implementation"). That said, if there's an expectation that spans are pointer followed by length, we may as well do that here.

Reply:
Hm I essentially duplicated the logic from the EBCO where the "empty" part comes first, using the pointer does indeed make sense too

Reply:
Moved to back